### PR TITLE
multistage docker builds make for slimmer images

### PR DESCRIPTION
**Pull request by mwalzer**
Adding a slimmer built using multistage docker build feature and one that has 3rd party binaries included.
Introducing labels and ('end') user change.
The folder/build name 'multistage' is probably subject to change, something along the lines of optimised. But I wanted to put the development so far out there, have some conversation as this PR evolves.

### CODE REVIEW

**Comment by mwalzer** [comment]
The image size for a working tools container is 435MB (559MB with OS installations of base dependencies)

File: batteries-included/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+RUN for l in lib/*.so; do strip --strip-all $l; done
+
+RUN mkdir /opt/thirdparty && \
+    svn checkout https://github.com/OpenMS/THIRDPARTY/trunk/Linux/64bit /opt/thirdparty && \
+    rm -rf /opt/thirdparty/.svn && \
+    svn checkout https://github.com/OpenMS/THIRDPARTY/trunk/All /opt/thirdparty && \
+    rm -rf /opt/thirdparty/.svn
+
+ENV PATH /opt/thirdparty/Fido:/opt/thirdparty/LuciPHOr2:/opt/thirdparty/MSGFPlus:/opt/thirdparty/MyriMatch:/opt/thirdparty/OMSSA:/opt/thirdparty/XTandem:$PATH
```
Comment (jpfeuffer) [inline batteries-included/Dockerfile]: We have more Thirdparties by now. If it should be up2date with develop, please add Sirius, SpectraST, Comet and Percolator.

Comment (jpfeuffer) [reply batteries-included/Dockerfile]: Ah I see you did that on the final container. Consider doing it here, too.

**Comment by mwalzer** [comment]
Regarding making multistage and batteries-included one build depends basically on our decision what release structure we want to follow. 
My suggestion would be to take the slimmest container without experimental optimization and make it 'the' OpenMS container (https://hub.docker.com/r/openms/openms/). And I think also that we should ship the 3rd party stuff with that build.
And then another image that still has everything contained as developer build instead of the executables image (I think the name is a bit unlucky). Unsure if we should go for 3rd party stuff inclusion there.
And then of course contrib, library, and pyopenms images

**Comment by jpfeuffer** [comment]
I actually meant to make them build/depend on each other instead of "making them one".
I agree that for the release build we should ship thirdparties.
Not sure if we need a developer build with executables but without thirdparties (maybe to have less licenses in there, so yes maybe we should add one).
The naming is up for discussion. If you have good suggestions, post them here or in a new issue.

**Pull request closed without merging**